lib/webpush/request.rb

```diff
@@ -36,52 +37,110 @@ def headers
       headers["Content-Type"] = "application/octet-stream"
       headers["Ttl"]          = ttl
 
-      if encrypted_payload?
+      if @payload.has_key?(:server_public_key)
         headers["Content-Encoding"] = "aesgcm"
         headers["Encryption"] = "salt=#{salt_param}"
         headers["Crypto-Key"] = "dh=#{dh_param}"
       end
 
-      headers["Authorization"] = "key=#{api_key}" if api_key?
+      if api_key?
```

Comment:
@zaru Yes, thank you! I'm sure there's a lot to improve on here, especially with request error handling, but I hope the community will find this changeset useful as a starting point with VAPID.